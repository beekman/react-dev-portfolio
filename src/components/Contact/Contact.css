@value colors: '../colors.css';
@value offwhite, white, whiteblue, lightblue, yellow, brightyellow, lightyellow, yellowgreen,  pink, bubblegum, green, darkyellow, darkgray, darkblue from colors;

.Contact{
  background-color: black;
  color: white;
  display: flex;
  flex-direction: column;
  justify-content: center;
  min-height: 100vh;
  align-items: center;
  font-size: 1.5rem;
  padding: 2rem;

  h1 {
    text-shadow: 0 0 0.075rem offwhite, 0 0 0.125rem offwhite, 0 0 0.15rem whiteblue, 0 0 0.25rem lightblue, 0 0 0.4rem lightblue, 0 0 0.5rem lightblue, 0 0 0.75rem lightblue, 0 0 1rem green;
    color: rgb(130, 134, 136)134, 136);
  }

  .hiringStatus {
    max-width: 60%;
  }

  .tagline {
    color: yellow;
    background-color: darkgray;
  }

  .social {
    display: block;
    text-align: left;
    max-width: 480px;
  }
}
