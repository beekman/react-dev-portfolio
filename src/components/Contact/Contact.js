import React from 'react';
import PropTypes from 'prop-types';
import styles from '../App.css';

const Contact = ({ contact }) => {
  return (
    <div className={ styles.Contact }>
      <h2>Contact Details</h2>
      <p className="address">
        <span>{ contact.name }</span><br />
        <span>{ contact.street }<br />
          { contact.city }, { contact.state } { contact.zip }
        </span>
        <span>{ contact.phone }</span><br />
        <span>{ contact.email }</span>
      </p>
    </div>
  );
};

Contact.propTypes = {
  contact: PropTypes.shape({
    name: PropTypes.string.isRequired,
    street: PropTypes.string.isRequired,
    city: PropTypes.string.isRequired,
    state: PropTypes.string.isRequired,
    zip: PropTypes.string.isRequired,
    phone: PropTypes.string.isRequired,
    email: PropTypes.string.isRequired
  })
};

export default Contact;
