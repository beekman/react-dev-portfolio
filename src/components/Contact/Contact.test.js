import React from 'react';
import { shallow } from 'enzyme';
import Contact from './Contact';
import data from '../../portfolio-data.js';

describe('Contact component', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<Contact contact={data.contact} />);
    expect(wrapper).toMatchSnapshot();
  });
});
