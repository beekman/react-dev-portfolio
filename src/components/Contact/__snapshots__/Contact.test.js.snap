// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact component matches snapshot 1`] = `
<Element
  className="Contact"
  name="Contact"
>
  <div
    className="Contact"
  >
    <ScrollAnimation
      animateIn="tada"
      animateOnce={false}
      animatePreScroll={true}
      delay={0}
      duration={1}
      initiallyVisible={false}
      offset={150}
    >
      <h1>
        Let's Connect!
      </h1>
    </ScrollAnimation>
    <p
      className="hiringStatus"
    >
      You can reach me using any of the contact options below.
    </p>
    <Typist
      avgTypingDelay={70}
      className=""
      cursor={
        Object {
          "show": false,
        }
      }
      delayGenerator={[Function]}
      onCharacterTyped={[Function]}
      onLineTyped={[Function]}
      onTypingDone={[Function]}
      startDelay={0}
      stdTypingDelay={25}
    >
      <h2
        className="tagline"
      >
        import SoftwareEngineer from 'benbeekman.dev';
      </h2>
    </Typist>
    <section
      className="social"
    >
      Ben Beekman
      <br />
      Portland
      , 
      OR
      <br />
      <br />
      <a
        href="!mailtoben@benbeekman.dev"
      >
        <AiOutlineMail />
        ben@benbeekman.dev
      </a>
      <br />
      <a
        href="https://www.linkedin.com/in/beekman/"
      >
        <FaLinkedin />
        /in/beekman
      </a>
      <br />
      <a
        href="https://github.com/beekman"
      >
        <FaGithub />
        /beekman
      </a>
      <br />
      <a
        href="https://www.canva.com/design/DAFYjYgRh2U/-dWbni-s5-Vx3sctCYxPIA/view?utm_content=DAFYjYgRh2U&utm_campaign=designshare&utm_medium=link&utm_source=publishsharelink"
        rel="noopener noreferrer"
        target="_blank"
      >
        View my resume
      </a>
    </section>
  </div>
</Element>
`;
